### DISCUSSION TITLE
Poll: Are you using Soto in a production environment?

### DISCUSSION THREAD

**Discussion by adam-fowler** [soto-project/soto#493] [Show and tell]
The proposal for Soto to become one of the SSWG recommended package has just gone up. https://forums.swift.org/t/sswg-0016-soto-for-aws/41552. One of the requirements for the package to move out of Sandbox to Incubating is

- Document that it is being used successfully in production by at least three independent end users which, in the SSWG judgement, are of adequate quality and scope.

If you are using Soto in a production environment, can you attach a 🚀 below. If you are able, can you provide details of how you are using it in a separate comment, provide details of services used, number of requests a day etc

thank you in advance

**Comment by 0xTim** [comment]
I'm using this is production with 2 projects:

* One is an events media social network that interacts with S3, SNS and STS. S3 is used to upload images via Soto. SNS to send notifications to both iOS and Android as well as send to fan out notifications to different services. STS is used to provide temporary credentials to the clients to upload large files (> 10GB)
* The other one is a grocery store app that interacts with S3, SNS and SES. Images are uploaded to S3 via Soto. Notifications are sent to users via SNS and SES is used to send any emails

**Comment by jonnymacs** [comment]
Disclaimer I am a contributor on this project (mostly in 2018/19)

We use Soto on 3 projects in production - the first having been deployed in 2018 is still running

- 2 event services. One service accepts post data from mobile marketing partners and the other from financial services partners. Both use SotoKinesis putRecords to post to kinesis. These services handle > 3k events / second at peak and about 150 million requests / day.
- an elastic search feeder. The feeder uses SotoS3 multi part download to read data files from S3. Data is then fed into elastic search. The feeder runs 1x / day.

**Comment by stsandro** [comment]
I'm using Soto in two small projects:

1. An invoice generation service. It loads raw payment data and stores it in S3, generates an invoice PDF and sends it via SES. It is also connecting to AWS SecretsManager to securely retrieve sensitive information. 😉

2. The second project is simple API that stores posted JSON into files on S3 and once per week the files are processed by a lambda function, zipped, stored in another S3 bucket and an email is sent via SES.

3. Another simple API that serves configurations stored in S3 to mobile Apps.

Those are all fairly small and simple projects, but they are running in production. All future projects, as well as well as existing ones, that make use of AWS Lambda aim to use Swift as programming language and of course Soto to connect to AWS services (DynamoDB, SES, API Gateway, S3, SNS, ...).

**Comment by Yasumoto** [comment]
We're using it in production as well, it's a small tool that is [used to track internal loadtests](https://github.com/Yasumoto/emergency-stop). Some [more description in the talk from last year's conference](https://www.serversideswift.info/speakers/joe-smith) too!

**Comment by jdmcd** [comment]
We just switched to using Soto instead of a custom S3 implementation we pieced together via previous open source SSS AWS projects. It's been a dream, and the fact that it supports all AWS APIs out of the box is really wonderful. 

I unfortunately can't speak to exact user or request numbers but happy to discuss more in a DM if that's helpful (I'm jdmcd in the Vapor Discord as well).

**Comment by leogdion** [comment]
Used Soto in an AWS Lambda environment for a client. Specifically the client was using DynamoDB, KMS, Shared Secrets, and Cognito as well. The app is probably receiving 100s of requests per day but it's currently just released.
